lightning/src/util/events.rs

```diff
@@ -112,8 +112,12 @@ pub enum Event {
 		/// payment is to pay an invoice or to send a spontaneous payment.
 		purpose: PaymentPurpose,
 	},
-	/// Indicates an outbound payment we made succeeded (ie it made it all the way to its target
+	/// Indicates an outbound payment we made succeeded (i.e. it made it all the way to its target
 	/// and we got back the payment preimage for it).
+	///
+	/// Note for MPP payments: we generate this event as soon as one of the payment paths succeeds. In
+	/// the case of a buggy payee, this may be followed by a PaymentFailed event if subsequent path(s)
```

Comment:
I don't believe it can be *followed* by a PaymentFailed, but it could be preceded by. It may be worth mentioning that users SHOULD treat this as a full payment sent, and MUST NOT attempt to retry parts of the payment.

Reply:
> and MUST NOT attempt to retry parts of the payment.

I guess that might not be helpful since they probably retried already when they received the preceding `PaymentFailed`?

I'm mildly in favor of removing this comment entirely, but I'll amend with your suggestion for now :)

Reply:
> I guess that might not be helpful since they probably retried already when they received the preceding PaymentFailed?

Ah, true lol.

> I'm mildly in favor of removing this comment entirely, but I'll amend with your suggestion for now :)

I think its worth at least mentioning that you can get a `PaymentFailed` followed immediately by a `PaymentSent`, which is confusing, and we should make sure users are aware of how they should treat that (payment success).